tests/api/printf.cpp

```diff
@@ -128,17 +134,19 @@ TEST_F(WithCommandQueue, SimplePrintf) {
     ASSERT_STREQ(printf_buffer, message);
 }
 
-TEST_F(WithCommandQueue, TooLongPrintf) {
+TEST_F(WithCommandQueueNoSetUp, TooLongPrintf) {
+    std::string buffer = "";
     // each print takes 12 bytes (4 for the printf_id, and 2*4 for the 2 integer
     // to print) + 4 for the byte written counter
-    auto cfg1 =
-        CLVK_CONFIG_SCOPED_OVERRIDE(printf_buffer_size, uint32_t, 28, true);
-
-    temp_folder_deletion temp;
-    stdoutFileName = getStdoutFileName(temp);
-
-    int fd;
-    ASSERT_TRUE(getStdout(fd));
+    cl_context_properties properties[4] = {
+        CL_PRINTF_CALLBACK_ARM, (cl_context_properties)printf_callback,
+        CL_PRINTF_BUFFERSIZE_ARM, (cl_context_properties)28};
+    WithCommandQueue::SetUpWithContextProperties(
```

Comment:
Doing the set up from within the test is a bit weird. I'm not sure that's recommended practise with Googletest. We could have a dedicated fixture where printf is always configured or maybe even do this in the default one.

Reply:
It can easily be done for the callback, but the complex part is that the test needs to define the buffer size.